### ISSUE TITLE
Issues with tokens

### ISSUE THREAD

**Issue by dfalling** [AppDoctorIo/accesspass#36]
This could likely be a problem with my server, but I'm having a brutal time debugging it.  If I login and then make a request using the returned access_code, I get a 401 error.  Playing with the genserver in the REPL suggests the genserver will successfully validate the token once, then subsequent attempts will fail:

```elixir
iex> a = AccessPass.AccessToken.add("blah", %{})
"OTZjODA2YWMtNGIxOS00MjUzLWE0MjktZDVlN2JmYmY0MGFm"
iex> a = AccessPass.AccessTokenServer.check(a)
{:ok, %{}}
iex> a = AccessPass.AccessTokenServer.check(a)
{:error, "access token expired"}
```

Any suggestions for troubleshooting this?  I've had issues before with my genserver being out of sync with my app post upgrade, but this seems different since I can reproduce in a single REPL.

**Comment by jpiepkow** [comment]
sure, Are you using normal plug or extend plug? Also are you sending the returned access token in the access_token header?

**Comment by dfalling** [comment]
Yeah, with the recent improvements I reverted to the standard plugs.  The bizarre thing is this works locally but not my prod server.

**Comment by jpiepkow** [comment]
That is quite interesting. I am using accesspass with no issues on appdoctor.io
If you are using a reverse proxy like nginx can you make sure its not removing the access_token header.

**Comment by jpiepkow** [comment]
are you using nginx?

**Comment by jpiepkow** [comment]
If you are you may need to set the following in your server blocks. This is most likely your issue. I should probably post an issue to switch accesspass to using access-token style headers instead.

underscores_in_headers on;

**Comment by dfalling** [comment]
I'm not- I'm just running it on Gigalixir.  It's all worked before, but I made two changes and it hasn't worked since:
- bumping to AccessPass@0.65
- upgrading to Distillery 2.0.10.

I suspect the problem's an issue with Gigalixir and my Distillery upgrade but am having trouble diagnosing it.

**Comment by dfalling** [comment]
And that was totally it- nginx was swallowing the underscored headers.

**Comment by jpiepkow** [comment]
Most likely in the next version I am going to switch access_token to access-token and refresh_token to refresh-token. I will bump major version number because it will be a breaking change.

**Comment by dfalling** [comment]
I'm totally down with that change!

**Comment by dfalling** [comment]
Great, thanks!
